Source/OCMock/NSInvocation+OCMAdditions.m

```diff
@@ -88,10 +88,17 @@ - (void)retainObjectArgumentsExcludingObject:(id)objectToExclude
             {
                 if(OCMIsBlockType(argumentType))
                 {
-                    // block types need to be copied in case they're stack blocks
-                    id blockArgument = [argument copy];
-                    [retainedArguments addObject:blockArgument];
-                    [blockArgument release];
+                    if (!OCMIsBlockNoEscape(argument))
+                    {
+                        // Block types need to be copied in case they're stack blocks.
+                        // No escape blocks should not be retained though because their life time is
+                        // stack based, but copying and releasing the non-escaping block is a no-op.
+                        // https://reviews.llvm.org/rGdbfa453e4138bb977644929c69d1c71e5e8b4bee
+                        // If we retain them here we will crash.
+                        id blockArgument = [argument copy];
```

Comment:
Can you clarify? You state that copy/release is a no-op. The proposed change simply doesn't call copy/retain on these no-escape blocks. But if they were no-ops anyway, how does that prevent a crash?

Reply:
So the key is that the proposed change doesn't add them to the retainedArguments array. Updated the comment to be clearer.